battle-engine.js

```diff
@@ -2470,6 +2476,40 @@ Battle = (function () {
 		}
 		return statuses;
 	};
+	/**
+	 * Use this function to attach custom event handlers to a battle. See Battle#runEvent for
+	 * more information on how to write callbacks for event handlers.
+	 *
+	 * Try to use this sparingly. Most event handlers can be simply placed in a format instead.
+	 *
+	 *     this.on(eventid, callback)
+	 * will set the callback as an event handler for when eventid is called, with priority 0.
+	 *
+	 *     this.on(eventid, priority, callback)
+	 * will set the callback as an event handler for when eventid is called, with priority equal
+	 * to the given priority.
+	 */
+	Battle.prototype.on = function (eventid/* [, priority], callback*/) {
+		var callback, priority;
+		if (arguments.length < 2) return;
```

Comment:
<strike>An string because yolo</strike>

A `TypeError` is the normal practice. Or just an `Error`. Really, it doesn't matter which one.